AnkiDroid/src/main/java/com/ichi2/anki/scheduling/SetDueDateDialog.kt

```diff
@@ -0,0 +1,227 @@
+                }
+            }
+            negativeButton(R.string.dialog_cancel)
+            neutralButton(R.string.help) {
+                (requireActivity() as AnkiActivity).openUrl(R.string.link_set_due_date_help)
+            }
+
+            setView(R.layout.dialog_set_due_date)
+        }.apply {
+            show()
+
+            lifecycleScope.launch {
+                viewModel.isValidFlow.collect { isValid -> positiveButton.isEnabled = isValid }
+            }
+            // setup viewpager + tabs
+            val viewPager = findViewById<ViewPager2>(R.id.pager)!!
+            viewPager.adapter = DueDateStateAdapter(this@SetDueDateDialog)
+            val tabLayout = findViewById<TabLayout>(R.id.tab_layout)!!
+            TabLayoutMediator(tabLayout, viewPager) { tab: TabLayout.Tab, position: Int ->
+                SetDueDateViewModel.Tab.entries.first { it.position == position }
+                    .let { selectedTab ->
+                        tab.setIcon(selectedTab.icon)
+                    }
+            }.attach()
+            tabLayout.selectTab(tabLayout.getTabAt(0))
+
+            viewPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+                override fun onPageSelected(position: Int) {
+                    SetDueDateViewModel.Tab.entries.first { it.position == position }.let { selectedTab ->
+                        viewModel.currentTab = selectedTab
+                    }
+                    super.onPageSelected(position)
+                }
+            })
+
+            // setup checkbox
+            findViewById<MaterialCheckBox>(R.id.change_interval)!!.setOnCheckedChangeListener { _, isChecked ->
+                viewModel.updateIntervalToMatchDueDate = isChecked
+            }
+        }
+    }
+
+    companion object {
+        const val ARG_CARD_IDS = "ARGS_CARD_IDS"
+
+        @CheckResult
+        fun newInstance(cardIds: List<CardId>) = SetDueDateDialog().apply {
+            arguments = bundleOf(ARG_CARD_IDS to cardIds.toLongArray())
+        }
+    }
+
+    class DueDateStateAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
+        override fun createFragment(position: Int): Fragment {
+            return when (position) {
+                0 -> SelectSingleDateFragment()
+                1 -> SelectDateRangeFragment()
+                else -> throw IllegalStateException("invalid position: $position")
+            }
+        }
+
+        override fun getItemCount() = 2
+    }
+
+    class SelectSingleDateFragment : Fragment() {
+
+        private val viewModel: SetDueDateViewModel by activityViewModels<SetDueDateViewModel>()
+
+        override fun onCreateView(
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
+        ): View? {
+            return inflater.inflate(R.layout.set_due_date_single, container, false)
+        }
```

Comment:
Not sure if you had tried this:

```suggestion
    class SelectSingleDateFragment : Fragment(R.layout.set_due_date_single) {

        private val viewModel: SetDueDateViewModel by activityViewModels<SetDueDateViewModel>()
```

Reply:
I hadn't, thanks